consensus/spos/bls/subroundSignature.go

```diff
@@ -239,3 +239,31 @@ func (sr *subroundSignature) signaturesCollected(threshold int) (bool, int) {
 
 	return n >= threshold, n
 }
+
+func (sr *subroundSignature) waitAllSignatures() {
```

Comment:
it would be easier for this func to get a context with cancel - and that context can be with timeout. Calculated only once with the remainingTime func.

Reply:
Done